### DISCUSSION TITLE
add magic scaler

### DISCUSSION THREAD

**Discussion by jcupitt** [libvips/libvips#2155] [Ideas]
See: 

http://www.johncostella.com/magic

A description of the resize kernel used by facebook and instagram. It should be easy to add to libvips.

**Comment by mertalev** [comment]
This kernel looks very impressive. I'd love to be able to use it!

There's a reference C implementation of it provided in that page. Do you think this would be implemented separately in libvips, or would libvips use the reference implementation?

**Reply by jcupitt** [reply]
libvips has a kernel system, I expect we'd just add `magic` as another option alongside nearest, linear, cubic, mitchell, lanczos2, and lanczos3.

PRs very welcome, if anyone is able. The code would slot in here:

https://github.com/libvips/libvips/blob/master/libvips/resample/templates.h#L435-L476

**Reply by jcupitt** [reply]
... I made an enhancement issue for this.
